### PR TITLE
Adding improvements

### PR DESCRIPTION
**Pull request by zbot473**
1. Use sbd as the tokenizer: this has a better tokenizer.
2. Add an array option to textrank and frequency

### CODE REVIEW

**Comment by SwapnikKatkoori** [comment]
Hi @zbot473, thanks for the pull request 🙏 . Contributions are greatly appreciated. Is this a completed PR that's ready for review, or are you still working on it?

**Comment by zbot473** [comment]
The first two commits are completed. Ill change the PR to say adding features, since more might come.

**Comment by SwapnikKatkoori** [comment]
Awesome, sounds good. Let me know when it is complete and I'll take a look.

**Comment by zbot473** [comment]
You can merge this

**Comment by SwapnikKatkoori** [comment]
Can you test it using npm test. I believe some of the unit test will need to be changed to account for the new changes.

**Comment by caelinsutch** [comment]
Still hasn't been tested?

**Comment by zbot473** [comment]
I'll fix my code to fit the tests, and also fix some spelling mistakes.
Sorry for forgetting

**Comment by zbot473** [comment]
@SwapnikKatkoori I fixed my code for the tests. Please check it out.

File: src/Preprocesser.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
-const natural = require("natural");
 const WordPos = require("wordpos");
 const WeightedGraph = require('./WeightedGraph').WeightedGraph;
 
 class Preprocesser{
 	constructor(){
-		this.tokenizer = new natural.SentenceTokenizer(); 
+		this.tokenizer = require('sbd');
```
Comment (caelinsutch) [inline src/Preprocesser.js]: Should this go in the header? Seems like a style thing but I guess there's no ESlint

File: src/Preprocesser.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 	}
 
 	//This method takes in a paragraph and returns a list of the sentences in the paragraph.
 	paragraphToSentences(string_to_process){
 		try{
-			let result = this.tokenizer.tokenize(string_to_process);
+			var result = this.tokenizer.sentences(string_to_process, {});
```
Comment (caelinsutch) [inline src/Preprocesser.js]: This should be a const

File: src/Summarizer.js [hunk 3]
```diff
@@ -43,6 +43,22 @@ class Summarizer{
 		return result_string;
 	}
 
+	summaryToArray(sorted_sentences, clean_sentences){
```
Comment (caelinsutch) [inline src/Summarizer.js:46]: Could you write a jsdoc for this to help the documentation effort?

**Review by caelinsutch — CHANGES_REQUESTED** [review]

File: src/Preprocesser.js (hunk 1 above)
Comment (zbot473) [reply src/Preprocesser.js]: Yeah i wrote this code a really long time ago :/

File: src/Summarizer.js (hunk 3 above)
Comment (zbot473) [reply src/Summarizer.js:46]: These methods aren't directly accessed by the user, rather they are called by the `summarizeByRank`and `summarizeByFrequency` methods. What do you think I should really write for them?

**Comment by zbot473** [comment]
I fixed those code style issues you had asked for. I'm still not too sure what to do with jsDoc, so I just put a simple comment explaining the method, as it is not really used by the end user.

File: src/Summarizer.js (hunk 3 above)
Comment (caelinsutch) [reply src/Summarizer.js:46]: I was thinking for future developers so it's easier to see and understand the code, but a comment is perfect! Thank you :)

**Review by caelinsutch — APPROVED** [review]

**Comment by SwapnikKatkoori** [comment]
@zbot473 Everything looks good. Thanks for all the effort!

**Pull request merged**